src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -348,6 +363,11 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
                 //first, manage primary contact role if the primary contact has changed
                 if (needsManageOCR(newOpp, oldOpp, FIELD_NAME_PRIMARY_CONTACT)) {
                     managePrimaryOCR(newOpp, oldOpp, mapOppIdMapConIdOCR.get(newOpp.id), isIndividualAccount);
+
+                    // Create Relationship Soft Credits accordingly
+                    if (!primaryContactsToOpportunities.isEmpty()) {
+                        OPP_AutomatedSoftCreditsService.createRelationshipOCRs(primaryContactsToOpportunities);
```

Comment:
you are now doing inserts of OCRs within a loop.  that's not good.  the rest of this code is putting the OCR's to be updated/inserted/deleted on dmlWrapper to avoid multiple dml's at multiple times.  I think you need createRelationshipOCRs() to return a list of OCRs to be created, which you can then add to dmlWrapper.

Reply:
Yes, excellent catch. I'll update the code. Thanks!